addon/components/ember-flatpickr.js

```diff
@@ -62,9 +50,175 @@ export default TextField.extend(Flatpickr, {
    * @type {String}
    */
 
-  didInsertElement() {
-    this._super(...arguments);
-    this.set('field', this.element);
+  @action
+  onInsert(element) {
+    this.field = element;
     this.setupFlatpickr();
   }
-});
+
+  @action
+  onWillDestroy() {
+    this.field._flatpickr.destroy();
+  }
+
+  setupFlatpickr() {
+    const { date, onChange, wrap } = this.args;
+
+    // Require that users pass a date
+    assert(
+      '<EmberFlatpickr> requires a `date` to be passed as the value for flatpickr.',
+      date !== undefined
+    );
+
+    // Require that users pass an onChange
+    assert(
+      '<EmberFlatpickr> requires an `onChange` action or null for no action.',
+      onChange !== undefined
+    );
+
+    // Wrap is not supported
+    assert(
+      '<EmberFlatpickr> does not support the wrap option. Please see documentation for an alternative.',
+      wrap !== true
+    );
+
+    // Pass all values and setup flatpickr
+    run.scheduleOnce('afterRender', this, this._setFlatpickrOptions);
+  }
+
+  _setFlatpickrOptions() {
+    const fastboot = getOwner(this).lookup('service:fastboot');
+
+    if (fastboot && fastboot.isFastBoot) {
+      return;
+    }
+
+    const {
+      date,
+      appendDataInput,
+      getFlatpickrRef,
+      disabled = false,
+      ...rest
+    } = this.args;
+
+    this.flatpickrRef = flatpickr(this.field, {
+      defaultDate: date,
+      ...rest,
+    });
+
+    if (appendDataInput) {
+      this.field.setAttribute('data-input', '');
+    }
+
+    this._setDisabled(disabled);
+
+    if (getFlatpickrRef) {
+      getFlatpickrRef(this.flatpickrRef);
+    }
+  }
+
+  /**
```

Comment:
Looks like we have some leftover comments here. Is there a reason we removed the `onClose`, `onOpen`, and `onReady` methods? I believe these served as a noop when they were not passed, and served to document the methods for users.